types/snaply/index.d.ts

```diff
@@ -0,0 +1,21 @@
+declare function deepClone<T>(obj: T, seen?: WeakMap<object, any>): T;
+
+/**
+ * Determines if the provided `value` is classified as a `Function` object.
+ *
+ * @since 1.0.1
+ * @param {unknown} value - The value to be checked.
+ * @returns {boolean} `true` if the value is a function, otherwise `false`.
+ */
+declare function isFunction(value: unknown): boolean;
+
+/**
+ * Determines if the provided `value` is classified as an `object`.
+ *
+ * @since 1.0.1
+ * @param {unknown} value - The value to be checked.
+ * @returns {boolean} `true` if the value is an object, otherwise `false`.
+ */
+declare function isObject(value: unknown): boolean;
```

Comment:
It may be nice for `isFunction` and `isObject` to be [type predicates](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70387/files#r1759646466). 

Otherwise, looks good!